pkg/kubelet/kubelet_pods.go

```diff
@@ -342,7 +343,14 @@ func ensureHostsFile(fileName string, hostIPs []string, hostName, hostDomainName
 		// if Pod is not using host network, create a managed hosts file with Pod IP and other information.
 		hostsFileContent = managedHostsFileContent(hostIPs, hostName, hostDomainName, hostAliases)
 	}
-
+	readContent, err := os.ReadFile(fileName)
+	if err == nil {
+		oldCheckSum := sha512.Sum512_256(readContent)
```

Comment:
Why do we use checksums for comparison here?

Reply:
we only want to check whether file changed. use checksum to improve comparative efficiency

Reply:
```
bytes.Equal(f1, f2)
```
@mrunalp do you suggest that we can use `bytes.Equal` here?

Reply:
bytes.Equal should be enough in my opinion.